### PR TITLE
Fabo/hid support

### PR DESCRIPTION
**Pull request by faboweb**
Switched to WebHID on Windows and WebUSB on OSX/Linux
U2F is unreliable and doesn't work on Windows

### CODE REVIEW

File: src/cosmos-ledger.ts [hunk 1]
```diff
@@ -74,7 +76,22 @@ export default class Ledger {
     // assume well connection if connected once
     if (this.cosmosApp) return this
 
-    let transport = await TransportU2F.create(timeout * 1000)
+    let transport
+    if (isWindows(this.platform)) {
+      if (!navigator.hid) {
+        throw new Error(
+          "Your browser doesn't have HID enabled. Please enable 'Experimental Web Platform features' here: chrome://flags / brave://flags"
```
Comment (jbibla) [inline src/cosmos-ledger.ts]:
```suggestion
          "Your browser doesn't have HID enabled. Please enable 'Experimental Web Platform' features here: chrome://flags / brave://flags"
```

Comment (jbibla) [reply src/cosmos-ledger.ts]: can we make `chrome://flags` a link some how? 

also, what specific feature needs to be enabled? the list is very long and we can select the anchor tag for the specific flag that needs to be changed.

File: src/cosmos-ledger.ts [hunk 2]
```diff
@@ -101,13 +118,11 @@ export default class Ledger {
...
-    const response = await this.cosmosApp.deviceInfo()
-    this.checkLedgerErrors(response)
-    // if (appName.toLowerCase() === `dashboard`) {
-    //   throw new Error(`Please open the Cosmos app.`)
-    // }
-
     const appName = await this.getOpenApp()
+
+    if (appName.toLowerCase() === `dashboard`) {
+      throw new Error(`Please open the Cosmos app.`)
```
Comment (jbibla) [inline src/cosmos-ledger.ts]:
```suggestion
      throw new Error(`Please open the Cosmos Ledger app on your Ledger device.`)
```

File: src/cosmos-ledger.ts [hunk 3]
```diff
@@ -101,13 +118,11 @@ export default class Ledger {
...
+    }
     if (appName.toLowerCase() !== `cosmos`) {
       throw new Error(`Close ${appName} and open the Cosmos app`)
```
Comment (jbibla) [inline src/cosmos-ledger.ts]:
```suggestion
      throw new Error(`Please close ${appName} and open the Cosmos Ledger app on your Ledger device.`)
```

**Comment by jbibla** [comment]
so glad this is going in. thanks @faboweb.

File: src/cosmos-ledger.ts (hunk 1 above)
Comment (faboweb) [reply src/cosmos-ledger.ts]: the item was called like this ^^ designers...

File: src/cosmos-ledger.ts [hunk 4]
```diff
@@ -74,7 +78,32 @@ export default class Ledger {
...
     if (this.cosmosApp) return this
 
-    let transport = await TransportU2F.create(timeout * 1000)
+    const browser = getBrowser(this.userAgent)
+
+    let transport
+    if (isWindows(this.platform)) {
+      if (!navigator.hid) {
+        throw new Error(
+          `Your browser doesn't have HID enabled. Please enable the feature by visiting following URL: ${browser}://flags/#enable-experimental-web-platform-features`
```
Comment (jbibla) [inline src/cosmos-ledger.ts]:
```suggestion
          `Your browser doesn't have HID enabled. Please enable this feature by visiting: ${browser}://flags/#enable-experimental-web-platform-features`
```

File: src/cosmos-ledger.ts [hunk 5]
```diff
@@ -74,7 +78,32 @@ export default class Ledger {
...
+    }
+    // OSX / Linux
+    else {
+      try {
+        const { default: TransportWebUSB } = await import('@ledgerhq/hw-transport-webusb')
+        transport = await TransportWebUSB.create(timeout * 1000)
+      } catch (err) {
+        if (err.message.trim().startsWith('No WebUSB interface found for your Ledger device')) {
+          throw new Error(
+            "Couldn't connect to a Ledger. You have a Ledger plugged in? Maybe it is outdated. Lunie requires at least FW 1.5.5 or later. You can use Ledger Live to update."
```
Comment (jbibla) [inline src/cosmos-ledger.ts]:
```suggestion
            "Couldn't connect to a Ledger device. Please use Ledger Live to upgrade the Ledger firmware to version 1.5.5 or later."
```

**Pull request merged**